Geometry/MTDNumberingBuilder/plugins/DDDCmsMTDConstruction.cc

```diff
@@ -63,9 +69,86 @@ std::unique_ptr<GeometricTimingDet> DDDCmsMTDConstruction::construct(const DDCom
+  CmsMTDConstruction theCmsMTDConstruction;
+
+  std::vector<GeometricTimingDet*> subdet;
+
+  do {
+    GeometricTimingDet::GeometricTimingEnumType fullNode = theCmsMTDStringToEnum.type(fv.name());
+    GeometricTimingDet::GeometricTimingEnumType thisNode = theCmsMTDStringToEnum.type(fv.name().substr(0, 7));
+    size_t num = fv.geoHistory().size();
+
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("MTDNumbering") << "Module level = " << limit << " current node level = " << num << " "
+                                     << fv.name();
+#endif
+
+    if (fullNode == GeometricTimingDet::BTL || fullNode == GeometricTimingDet::ETL) {
+      limit = 0;
+
+      // define subdetectors as GeometricTimingDet components
+
+      subdet.emplace_back(theCmsMTDConstruction.buildSubdet(fv, mtd.get(), attribute));
+    }
+    //
+    // the level chosen corresponds to wafers for D50 and previous scenarios
+    //
+    if ((thisNode == GeometricTimingDet::BTLModule || thisNode == GeometricTimingDet::ETLModule) && limit == 0) {
+      limit = num + 1;
+    }
+    if (num != limit && limit > 0) {
+      continue;
+    }
+    if (thisNode == GeometricTimingDet::BTLModule) {
+      theCmsMTDConstruction.buildBTLModule(fv, subdet.back(), attribute);
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("MTDNumbering") << "Registered in GeometricTimingDet as type " << thisNode;
+#endif
+      limit = num;
+    } else if (thisNode == GeometricTimingDet::ETLModule) {
+      theCmsMTDConstruction.buildETLModule(fv, subdet.back(), attribute);
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("MTDNumbering") << "Registered in GeometricTimingDet as type " << thisNode;
+#endif
+      limit = num;
+    }
+  } while (fv.next());
+
+  // sort GeometricTimingDet
+
+#ifdef EDM_ML_DEBUG
+  auto comp = mtd->deepComponents();
+  std::stringstream before(std::stringstream::in | std::stringstream::out);
+  for (const auto& it : comp) {
+    before << "ORDER1 " << it->geographicalId().rawId() << " " << it->type() << " " << it->translation().z() << "\n";
+  }
+  edm::LogVerbatim("MTDNumbering") << "GeometricTimingDet order before sorting \n" << before.str();
+#endif
+
+  for (size_t index = 0; index < subdet.size(); index++) {
+    GeometricTimingDet::ConstGeometricTimingDetContainer& icomp = subdet[index]->components();
+    std::stable_sort(icomp.begin(), icomp.end(), CmsMTDConstruction::mtdOrderZ);
+    std::stable_sort(icomp.begin(), icomp.end(), CmsMTDConstruction::mtdOrderRR);
+    if (index > 0) {
+      std::stable_sort(icomp.begin(), icomp.end(), CmsMTDConstruction::mtdOrderPhi);
+    }
+  }
+
+#ifdef EDM_ML_DEBUG
+  comp.clear();
+  comp = mtd->deepComponents();
+  std::stringstream after(std::stringstream::in | std::stringstream::out);
+  for (const auto& it : comp) {
+    after << "ORDER2 " << it->geographicalId().rawId() << " " << it->type() << " " << it->translation().z() << "\n";
+  }
+  edm::LogVerbatim("MTDNumbering") << "GeometricTimingDet order before sorting \n" << after.str();
```

Comment:
should this message say "after sorting"?

Reply:
of course, a leftover of many cut-and-paste during debugging additions and removals